python/generate_syntax_highlightining.py

```diff
@@ -106,3 +109,50 @@ def match_all_variants(token_names):
 
 with open(filename, "w") as outfile:
     json.dump(data, outfile, indent=4)
+
+## Glossary files.  Reference:
+## http://oracc.museum.upenn.edu/doc/help/glossaries/index.html
+# Initialise the dictionary
+data = {}
```

Comment:
This may be overkill since we likely won't add more languages/syntaxes, but could we avoid the duplication here? Maybe load the keywords for each language from a YAML/JSON file or something? Or at least define a function that creates the `data` dictionary and writes it to a file, so we can call it with different contents for each "part"?

Reply:
Yeah, I wasn't super happy about the duplication here, but it isn't actually _that_ much, I think.

I'm not happy about reading from JSON, the point of this script is to programmatically generate the JSON files (admittedly more important for the ATF file where we use pyoracc): if I had to read the configuration from a JSON file, I wouldn't use this script in the first place :slightly_smiling_face: 

Anyway I'll into the possibility of reducing duplication where possible.

Reply:
Alright, I reworked the script to reduce duplication as much as possible, mainly when building the `data` and `repository` dictionaries :slightly_smiling_face: